mlem/core/objects.py

```diff
@@ -673,6 +674,9 @@ def write_value(self) -> Artifacts:
     def load_value(self):
         self.dataset = self.reader.read(self.relative_artifacts)
 
+    def read_batch(self, batch: int) -> Iterator[DatasetType]:
+        return self.reader.read_batch(self.relative_artifacts, batch)  # type: ignore
```

Comment:
why `type: ignore`?

Reply:
Ah, so `mypy` was complaining about this previously and I've circumvented this by an assertion check rather than ignoring. Will remove `type: ignore`.

```
mlem/core/objects.py: note: In member "read_batch" of class "DatasetMeta":
mlem/core/objects.py:685:16: error: Item "None" of "Optional[DatasetReader]"
has no attribute "read_batch"  [union-attr]
```